Composer/packages/client/src/components/WebChat/WebChatComposer.tsx

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import React from 'react';
+import { createStyleSet, Components } from 'botframework-webchat';
+import { CommunicationColors, NeutralColors } from '@uifabric/fluent-theme';
+
+import { ConversationService } from './utils/conversationService';
+import webChatStyleOptions from './utils/webChatTheme';
+import { ChatData, ActivityType } from './types';
+const { BasicWebChat, Composer } = Components;
+
+export type WebChatComposerProps = {
+  currentConversation: string;
+  activeLocale: string;
+  conversationService: ConversationService;
+  botUrl: string;
+  chatData: ChatData;
+  isDisabled: boolean;
+};
+
+const createCardActionMiddleware = () => (next) => async ({ cardAction, getSignInUrl }) => {
+  const { type, value } = cardAction;
+
+  switch (type) {
+    case 'signin': {
+      // eslint-disable-next-line security/detect-non-literal-fs-filename
+      const popup = window.open();
+      const url = await getSignInUrl();
+      if (popup) {
+        popup.location.href = url;
+      }
+      break;
+    }
+
+    case 'downloadFile':
+    //Fall through
+
+    case 'playAudio':
+    //Fall through
+
+    case 'playVideo':
+    //Fall through
+
+    case 'showImage':
+    //Fall through
+
+    case 'openUrl':
+      // eslint-disable-next-line security/detect-non-literal-fs-filename
+      window.open(value, '_blank');
+      break;
+
+    default:
+      return next({ cardAction, getSignInUrl });
+  }
+};
+
+const createActivityMiddleware = () => (next: unknown) => (...setupArgs) => (...renderArgs) => {
+  const card = setupArgs[0];
+  switch (card.activity.type) {
+    case ActivityType.Trace:
+      return false;
+
+    case ActivityType.EndOfConversation:
+      return false;
+
+    default:
+      if (typeof next === 'function') {
+        const middlewareResult = next(...setupArgs);
+        if (middlewareResult) {
+          return middlewareResult(...renderArgs);
+        }
+        return false;
+      }
+      return false;
+  }
+};
+
+const areEqual = (prevProps: WebChatComposerProps, nextProps: WebChatComposerProps) => {
+  const result =
+    prevProps.currentConversation === nextProps.currentConversation && prevProps.isDisabled === nextProps.isDisabled;
+  return result;
+};
+
+export const WebChatComposer = React.memo((props: WebChatComposerProps) => {
+  const { activeLocale, chatData, isDisabled, currentConversation } = props;
+
+  if (!currentConversation || !chatData) {
+    return null;
+  }
+
+  const styleSet = createStyleSet({ ...webChatStyleOptions });
```

Comment:
can this be moved outside the component to get called once?

Reply:
I did try this before. On every instantatiation of webchat it keeps resetting back to the old style. Hence dropped it here.